temporian/core/data/sampling.py

```diff
@@ -26,27 +26,30 @@
     Union,
 )
 
-from temporian.core.data.dtype import DType
+from temporian.core.data.dtype import DType, py_types_to_dtypes
 
 if TYPE_CHECKING:
     from temporian.core.operators.base import Operator
 
 
+class IndexLevel(NamedTuple):
+    name: str
+    dtype: IndexDType
+
+
 class IndexDType(Enum):
     INT32 = DType.INT32.value
     INT64 = DType.INT64.value
     STRING = DType.STRING.value
 
 
-class IndexLevel(NamedTuple):
-    name: str
-    dtype: IndexDType
+AnyDType = Union[DType, IndexDType, type]
 
 
 class Index:
     def __init__(
         self,
-        levels: Union[Tuple[str, IndexDType], List[Tuple[str, IndexDType]]],
+        levels: Union[Tuple[str, AnyDType], List[Tuple[str, AnyDType]]],
```

Comment:
an index's type cannot be a float, so using `AnyDType` is too permissive. If we need to support python dtypes we would need to allow `IndexDType` + [`int`, `str`]